source/particles/generators.cc

```diff
@@ -391,6 +397,73 @@ namespace Particles
         particle_handler.insert_particles(particles);
       }
     }
+
+    template <int dim, int spacedim>
+    void
+    non_matching_dof_support_points(
+      const DoFHandler<dim, spacedim> &particle_dof_handler,
```

Comment:
I'd rename this argument `dof_handler`. It may or may not be associated with any particles.